### DISCUSSION TITLE
Accepted way to timeout a request?

### DISCUSSION THREAD

**Discussion by abarrettPS** [zino-app/graphql-flutter#1225] [Q&A]
If I make a request in a `build()` method using `Query`, how do I set a timeout on the request? I only want to try fetching for a few seconds before alerting the UI to show an error.

What I'm doing:

```
  @override
  Widget build(BuildContext context) {
    return Query(
      options: QueryOptions(
          fetchPolicy: FetchPolicy.cacheAndNetwork,
          document: gql(queryString),
          variables: variables ?? const {},
          parserFn: dataParser),
      builder: (QueryResult result,
          {VoidCallback? refetch, FetchMore? fetchMore}) {
...
```

**Comment by vincenzopalazzo** [comment]
I never had this problem, and I'm happy to make some research into it. In the meanwhile feel free to explore the code and update the answer with the solution found
